packages/shared/constants/src/lib/infrastructure.ts

```
@@ -0,0 +1,121 @@
+// everywhere. Follows the same `as const satisfies Record<...>` pattern as
+// MODE_PROJECT_MAP in project.ts.
+//
+// D1 database identities (per-mode): the hub's Cloudflare D1 binding.
+// R2 bucket identities (per-mode): SAVES_BUCKET (save backups + user objects)
+// and CATALOG_BUCKET (catalog asset origin).
+//
+// Three call sites can't import TS and must stay in sync by convention:
+// wrangler.jsonc (static JSONC), .pi extensions (no moon project graph), and
+// bash scripts (scripts/direnv/bootstrap.sh).
+
+import { modes } from './project.ts';
```

Biome (2.5.8)

[error] All these imports are only used as types.

(lint/style/useImportType)